### ISSUE TITLE
Cannot save Jpeg2000 image with integer data 

### ISSUE THREAD

**Issue by sidy3d** [cgohlke/imagecodecs#101]
Hi,
I was recently experimenting with this library to save lossless jpeg2000 images. It seems that this functionality is broken.

Example code:
```
imagecodecs.jpeg2k_encode(np.random.randint(0, 255, (256, 256), np.uint8), level=0)
```

Error message:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "imagecodecs\_jpeg2k.pyx", line 406, in imagecodecs._jpeg2k.jpeg2k_encode
imagecodecs._jpeg2k.Jpeg2kError: opj_end_compress failed
```

**Comment by cgohlke** [comment]
Bummer. The `jpeg2k_encode` function writes to a pre-allocated buffer, which is too small for some cases. It should write to a stream instead... To work around this issue for now, specify a large output buffer, something like `jpeg2k_encode(im, out=max(im.nbytes * 2, 4096)`.

**Comment by sidy3d** [comment]
Thanks for the workaround. I am trying to use jpeg2000 compression and TIFF files through the tiffile library, any workarounds for that usecase?

**Comment by cgohlke** [comment]
> any workarounds for that usecase?

Use the `compressionargs`parameter to pass the output buffer size to the encoder, for example:

```python
import numpy
import tifffile

im = numpy.random.randint(0, 255, (512, 512, 3), numpy.uint8)

tifffile.imwrite('issue101.tif', im, tile=(256, 256), compression='jpeg2000', compressionargs={'out': 256*256*3*2})
```

**Comment by sidy3d** [comment]
> > any workarounds for that usecase?
> 
> Use the `compressionargs`parameter to pass the output buffer size to the encoder, for example:
> 
> ```python
> import numpy
> import tifffile
> 
> im = numpy.random.randint(0, 255, (512, 512, 3), numpy.uint8)
> 
> tifffile.imwrite('issue101.tif', im, tile=(256, 256), compression='jpeg2000', compressionargs={'out': 256*256*3*2})
> ```

That seems to work. Thank you very much.

Not sure if I should leave the issue open?

**Comment by cgohlke** [comment]
Thanks for reporting the issue! Yes, leave it open for now. It should be fixed in the next release.

**Comment by cgohlke** [comment]
Fixed in v2024.6.1.

**Comment by sidy3d** [comment]
Can confirm that the new version works. Thanks!
